tensorflow/contrib/learn/python/learn/estimators/base.py

```diff
@@ -188,7 +189,7 @@ def _setup_summary_writer(self, logdir):
             os.path.join(logdir, datetime.datetime.now().strftime('%Y-%m-%d_%H-%M-%S')),
             graph=self._session.graph)
 
-    def fit(self, X, y, monitor=None, logdir=None):
+    def fit(self, X, y=None, monitor=None, logdir=None):
```

Comment:
hm, not sure if that's the best option - but we can create our own UnsupervisedMixin that will have this kind of fit(X, ...) and may be some other functionality later.